sources/editor/Stride.Assets.Presentation/TemplateProviders/EntityComponentReferenceTemplateProvider.cs

```diff
@@ -13,7 +15,13 @@ public class EntityComponentReferenceTemplateProvider : NodeViewModelTemplatePro
 
         public override bool MatchNode(NodeViewModel node)
         {
-            return typeof(EntityComponent).IsAssignableFrom(node.Type) && node.Parent?.Type != typeof(EntityComponentCollection);
+            if (node.Parent?.Type == typeof(EntityComponentCollection))
+                return false;
+
+            if (typeof(EntityComponent).IsAssignableFrom(node.Type))
+                return true;
+
+            return node.Type.IsInterface && node.Type.IsImplementedOnAny<EntityComponent>();
```

Comment:
Although, I just noticed - in your demo you show that when a non-component type is chosen it still shows the entity picker - can you see if it's possible to filter it out? 
Maybe we could check something more about `node`'s value type here?

Reply:
Initially the idea was to remove reference types from the combo box and only use the picker to select them.

Issue is that would prevent users from creating new instances of those component types inline to the field (i.e.: like a value type instead of ref) - this probably wouldn't have been a very common way to use this feature, but might as well account for it. Now it worked before but it looks like I ended up breaking it somewhere along the line.

Anyway, even without that particular feature working, the picker icon is still a shortcut to select any instance of the interface, it's not constrained to the type that was selected in the combo box, so it's one less click for users if they want to swap from an inlined instance to a ref.

Reply:
Oh, I see. I get that when we're asking for the interface instance we don't exactly care about the implementing type. But if we're anyways making the user choose some type I'd expect the entity picked to be then constrained to the type I have chosen (especially with bigger scenes). Would it be possible to delegate the picker to act on the chosen type instead - this way we don't have to add all that interface logic to entity picker.

Reply:
Right, I don't think the picker should by default constrain based on the selected type because this would be very easy to miss from the user's perspective. We don't have any UX setup which would make this idea obvious every time a user would use the picker. It would also be kind of awful for users to go back and forth between the combo box and the picker if they selected the wrong type by mistake.

Filtering is a good point though, but I think this should be part of the object picker window itself, that way users can benefit from those filters for direct component properties/fields, and there's so much dead space in that window it would be a shame not to use it ?

Reply:
Sounds good ! I'm really bad at xaml though, so let's get this in and see if anyone can take over ?

Reply:
Go with it. Hopefully someone knowledgeable in XAML can implement the filtering. I'm not very proficient in XAML but if no one looks at this, I'll take a look later.

Reply:
Ok, let's merge it. And I'll open an enhancement issue on UX of abstract types in lists in general - I think when it's a single interface property the drop-down holds the selected type name.